### PR TITLE
docs: update invalid link

### PR DESCRIPTION
**Pull request by hoanguyenkh**
## Summary by CodeRabbit

- **Documentation**
  - Updated documentation links across various modules to reflect recent changes in the codebase.
- **New Features**
  - Enhanced transaction handling in the `BlockExecutor` and `mempool` module for improved performance.
- **Bug Fixes**
  - Adjusted links in documentation to correct line numbers for better accuracy and user guidance.
- **Refactor**
  - Removed specific fields in `Header` struct to enforce new behaviors in upcoming versions.

### CODE REVIEW

File: p2p/p2p.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ func (ln *LightNode) falseValidator() p2p.GossipValidator {
 
 [4] [conngater][conngater]
 
-[client.go]: https://github.com/rollkit/rollkit/blob/main/p2p/client.go#L43
+[client.go]: https://github.com/rollkit/rollkit/blob/main/p2p/client.go#L51
```
Comment (MSevey) [inline p2p/p2p.md:57]: this one should just drop the line number and just reference the file.

File: state/indexer-service.md [hunk 2]
```diff
@@ -56,5 +56,5 @@ See [indexer service]
 
 [block_indexer]: https://github.com/rollkit/rollkit/blob/main/state/indexer/block.go#L11
 [tx_indexer]: https://github.com/rollkit/rollkit/blob/main/state/txindex/indexer.go#L14
-[publish_events_method]: https://github.com/rollkit/rollkit/blob/main/state/executor.go#L352
-[indexer service]: https://github.com/rollkit/rollkit/blob/main/state/txindex/indexer_service.go#L20
+[publish_events_method]: https://github.com/rollkit/rollkit/blob/main/state/executor.go#L353
```
Comment (MSevey) [inline state/indexer-service.md:59]: this isn't pointing to the publish events method, should be L409

File: state/indexer-service.md [hunk 3]
```diff
@@ -56,5 +56,5 @@ See [indexer service]
...
+[indexer service]: https://github.com/rollkit/rollkit/blob/main/state/txindex/indexer_service.go#L19
```
Comment (MSevey) [inline state/indexer-service.md:60]: This one can remove the line number as well for easier maintenance.

File: types/block_spec.md [hunk 4]
```diff
@@ -75,22 +75,22 @@ Block.Verify()
...
+## [SignedHeader](https://github.com/rollkit/rollkit/blob/main/types/signed_header.go#L15)
 
 | **Field Name** | **Valid State**                                                          | **Validation**                                                                              |
 |----------------|--------------------------------------------------------------------------|---------------------------------------------------------------------------------------------|
 | Header         | Valid header for the block                                               | `Header` passes `ValidateBasic()` and `Verify()`                                            |
 | Commit         | 1 valid signature from the centralized sequencer                             | `Commit` passes `ValidateBasic()`, with additional checks in `SignedHeader.ValidateBasic()` |
 | Validators     | Array of Aggregators, must have length exactly 1. | `Validators` passes `ValidateBasic()`                                                       |
 
-## [Header](https://github.com/rollkit/rollkit/blob/main/types/header.go#L25)
+## [Header](https://github.com/rollkit/rollkit/blob/main/types/header.go#L29)
```
Comment (MSevey) [inline types/block_spec.md:93]: This should be L39 for the `Header` as the `BaseHeader` is referenced below.

**Review by MSevey — CHANGES_REQUESTED** [review]
Thanks for the PR. Left some additional updates.

**Comment by hoanguyenkh** [comment]
thank @MSevey for feedback, i updated

**Review by MSevey — APPROVED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]

**Pull request merged**